src/interactive/plots.ts

```diff
@@ -26,6 +28,120 @@ export function activate(context: vscode.ExtensionContext) {
     context.subscriptions.push(vscode.commands.registerCommand('language-julia.plotpane-delete', plotPaneDel))
 
     context.subscriptions.push(vscode.commands.registerCommand('language-julia.plotpane-delete-all', plotPaneDelAll))
+
+    context.subscriptions.push(vscode.commands.registerCommand('language-julia.show-plot-navigator', g_plotProvider.showPlotNavigator))
+
+    vscode.window.registerWebviewViewProvider('julia-plot-navigator', g_plotProvider)
+}
+
+interface Plot {
+    thumbnail_type: string,
+    thumbnail_data: string
+}
+
+class PlotViewProvider implements vscode.WebviewViewProvider {
+    private view?: vscode.WebviewView
+    private plotsInfo?: Array<Plot>
+
+    constructor() {
+    }
+
+    resolveWebviewView(view: vscode.WebviewView, context: vscode.WebviewViewResolveContext) {
+        this.view = view
+
+        this.plotsInfo = []
+
+        view.webview.options = {
+            enableScripts: true,
+            enableCommandUris: true
+        }
+
+        view.webview.onDidReceiveMessage(msg => {
+            // msg.type could be used to determine messages
+            switch (msg.type) {
+                case "toPlot":
+                    if (msg.value >= 0 && msg.value <= g_plots.length - 1) {
+                        g_currentPlotIndex = msg.value
+                        updatePlotPane()
+                    }
+                    break
+                default:
+                    console.error(`Unknown message type from WebView: ${msg.type}, value: ${msg.value}`)
+            }
+
+        })
+
+        this.reloadPlotPane();
+    }
+
+    getWebviewHTML(innerHTML: string) {
+        const extensionPath = g_context.extensionPath
```

Comment:
I'd prefer having `g_context` be a property of this class (set in the constructor) instead of referencing the global. That's basically the direction we wanna go everywhere in the extension.

Reply:
Fixed.